redfish-core/lib/log_services.hpp

```diff
@@ -3346,4 +3415,931 @@ inline void requestRoutesPostCodesEntry(App& app)
+ *
+ * @return The redfish response with appropriate redfish properties of the
+ *         isolated hardware details into LogEntry schema if success else
+ *         nothing in redfish response.
+ */
+inline void fillIsolatedHwDetailsByObjPath(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const sdbusplus::message::object_path& dbusObjPath,
+    const size_t entryJsonIdx)
+{
+    // Fill Redfish uri of isolated hardware into "OriginOfCondition"
+    if (dbusObjPath.filename().find("unit") != std::string::npos)
+    {
+        // If Isolated Hardware object name contain "unit" then that unit
+        // is not modelled in inventory and redfish so the "OriginOfCondition"
+        // should filled with it's parent (aka FRU of unit) path.
+        getRedfishUriByDbusObjPath(asyncResp, dbusObjPath.parent_path(),
+                                   entryJsonIdx);
+    }
+    else
+    {
+        getRedfishUriByDbusObjPath(asyncResp, dbusObjPath, entryJsonIdx);
+    }
+
+    // Fill PrettyName of isolated hardware into "Message"
+    getPrettyNameByDbusObjPath(asyncResp, dbusObjPath, entryJsonIdx);
+}
+
+/**
+ * @brief API used to fill isolated hardware details into LogEntry schema
+ *        by using the given isolated dbus object.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details. If passing less than or equal 0 then,
+ *                            it will assume the given asyncResp jsonValue as
+ *                            a single entry json object. If passing greater
+ *                            than 0 then, it will assume the given asyncResp
+ *                            jsonValue contains "Members" to fill in the
+ *                            appropriate entry json object.
+ * @param[in] dbusObjIt - The DBus object which contains isolated hardware
+                         details.
+ *
+ * @return The redfish response with appropriate redfish properties of the
+ *         isolated hardware details into LogEntry schema if success else
+ *         failure response.
+ */
+inline void fillSystemHardwareIsolationLogEntry(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const size_t entryJsonIdx, GetManagedObjectsType::const_iterator& dbusObjIt)
+{
+    nlohmann::json& entryJson =
+        (entryJsonIdx > 0
+             ? asyncResp->res.jsonValue["Members"][entryJsonIdx - 1]
+             : asyncResp->res.jsonValue);
+
+    bool entryIsPresent{false};
+    for (auto& interface : dbusObjIt->second)
+    {
+        if (interface.first == "xyz.openbmc_project."
+                               "HardwareIsolation.Entry")
+        {
+            entryIsPresent = true;
+            for (auto& property : interface.second)
+            {
+                if (property.first == "Severity")
+                {
+                    if (auto severity =
```

Comment:
don't use auto here and please check nullptr

Reply:
> don't use auto here

Can I know the reason "why we cannot use auto here"?

> please check nullptr

Not required, since the implicit conversion will happen that will return true for non-null pointer and false for a null pointer.